pkg/reconciler/reconciler_test.go

```diff
@@ -1198,9 +1199,10 @@ var _ = Describe("Reconciler", func() {
 								})
 
 								By("successfully reconciling a request", func() {
-									res, err := r.Reconcile(ctx, req)
-									Expect(res).To(Equal(reconcile.Result{}))
-									Expect(err).To(BeNil())
+									Eventually(func() bool {
+										result, err := r.Reconcile(ctx, req)
+										return result == reconcile.Result{} && err == nil
+									}).Should(BeTrue())
```

Comment:
Hey, thanks for the comments. It seems that when uninstalling, it will always lead to a conflict. I was unable to locate where this conflict actually comest from, but I suspect https://github.com/operator-framework/helm-operator-plugins/blob/c16a400954f7e43bb987b196e8ecf2f4d2d4ab0f/pkg/reconciler/reconciler.go#L674

Reply:
> It seems that when uninstalling, it will always lead to a conflict.

Oh bummer, we should fix that. I don't see anything there that's immediately obvious other than that it calls `updater.Apply`. I wonder if it is operating on a stale cache? Any chance you have the time to try to debug further (e.g. add some extra context to the errors you suspect are resulting in the conflict) to see if we can nail it down?

Reply:
Hey :) I think I might've found it. I believe that the deletion handler waits for the resource to be deleted, but the closure in the main reconciliation loop will attempt to update the resource after it was deleted.